src/t8_forest/t8_forest_ghost.cxx

```diff
@@ -303,6 +303,55 @@ t8_forest_ghost_tree_num_elements (t8_forest_t forest, t8_locidx_t lghost_tree)
   return t8_element_array_get_count (&ghost_tree->elements);
 }
 
+const t8_element_t *
+t8_ghost_get_ghost_in_tree (t8_forest_t forest, t8_locidx_t lghost_tree, t8_linearidx_t linear_id, int element_level,
+                            t8_locidx_t *loc_ghost_id)
+{
+  T8_ASSERT (t8_forest_is_committed (forest));
+  const t8_element_array_t *ghost_elements = t8_forest_ghost_get_tree_elements (forest, lghost_tree);
+  T8_ASSERT (ghost_elements != NULL);
+  /* Search for the element.
+   * The search returns the largest index i,
+   * such that the element at position i has a smaller id than the given one.
+   * If no such i exists, it returns -1. */
+  *loc_ghost_id = t8_forest_bin_search_lower (ghost_elements, linear_id, element_level);
+  if (*loc_ghost_id >= 0) {
+    /* The element was found */
+    return (const t8_element_t *) t8_sc_array_index_locidx (&(ghost_elements->array), *loc_ghost_id);
+  }
+  return nullptr;
+}
+
+t8_locidx_t
+t8_ghost_get_ghost_id_in_tree (t8_forest_t forest, t8_locidx_t lghost_tree, t8_element_t *ghost_element)
+{
+  T8_ASSERT (t8_forest_is_committed (forest));
```

Comment:
Add an assertion that the forest is not transitioned/does not have subelements.

The current version does not work with subelements since all subelements of the same elements have the same linear id, thus the search may find a different element, even if it exists.

We can either extend this search or (if there is not time) keep it but abort if it is used with subelements.